### PR TITLE
Added non-allocating Base64 encoder/decoder

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
This is not the fastest implementation I can imagine, but at least it's non-allocating, i.e. operated passed in byte buffers.

### CODE REVIEW

**Comment by KrzysztofCwalina** [comment]
as requested by @davidfowl

**Comment by jonfortescue** [comment]
This LGTM

**Comment by davidfowl** [comment]
System.Binary.Base64, a whole new package for this single class with 2 functions? NPM here we come 😄 .

File: src/System.Binary.Base64/project.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "licenseUrl": "http://go.microsoft.com/fwlink/?LinkId=329770",
+    "iconUrl": "http://go.microsoft.com/fwlink/?LinkID=288859",
+    "projectUrl": "https://github.com/dotnet/corefxlab",
+    "requireLicenseAcceptance": true,
+  },
+  "buildOptions": {
+    "keyFile": "../../tools/Key.snk"
+  },
+  "dependencies": {
+    "System.Runtime": "4.0.0",
```
Comment (davidfowl) [inline src/System.Binary.Base64/project.json]: This should be 4.1.0

**Comment by KrzysztofCwalina** [comment]
I might put it into System.Binary later. I just need to remove dependency on Utf8String

File: src/System.Binary.Base64/project.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "System.Runtime.InteropServices": "4.0.0",
+    "System.Diagnostics.Debug": "4.0.0",
```
Comment (davidfowl) [inline src/System.Binary.Base64/project.json]: 4.0.11

File: src/System.Binary.Base64/project.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "iconUrl": "http://go.microsoft.com/fwlink/?LinkID=288859",
+    "projectUrl": "https://github.com/dotnet/corefxlab",
+    "requireLicenseAcceptance": true,
+  },
+  "buildOptions": {
+    "keyFile": "../../tools/Key.snk"
+  },
+  "dependencies": {
+    "System.Runtime": "4.0.0",
+    "System.Runtime.InteropServices": "4.0.0",
```
Comment (davidfowl) [inline src/System.Binary.Base64/project.json]: 4.1.0

File: src/System.Binary.Base64/project.json (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.Binary.Base64/project.json]: Why?

**Comment by davidfowl** [comment]
@KrzysztofCwalina WRT why the later versions of dependencies:
The latest package versions contain *older* DLL versions. Targeting netstandard 1.0 is enough to run *everywhere*. If you don't use the packages that have netstandard then you're forcing all consumers to use "imports" because older package versions don't have the netstandard moniker.

**Comment by davidfowl** [comment]
Also do these get published on every commit or just once daily?

**Comment by KrzysztofCwalina** [comment]
@davidfowl, I will merge and kick start package build.

**Comment by KrzysztofCwalina** [comment]
I would love to target NS 1.0, but not without trimming. A few months ago somebody changed the dependencies to NS1.0 and it was a mess. Programs would not compile because there was something wrong with packages that the program was not even using (but are part of NS 1.0). And so I change it back to use individual contracts.

I don't want to target newer individual contracts, because I only use APIs available in S.R 4.0, and so why would I (if I don't use NS 1.0).

**Comment by KrzysztofCwalina** [comment]
The package is now on https://dotnet.myget.org/feed/dotnet-corefxlab/package/nuget/System.Binary.Base64

**Pull request merged**